src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionaryUtility.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Framework.Internal;
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    /// <summary>
+    /// Utility class for clearing entries in <see cref="ModelStateDictionary"/>.
+    /// </summary>
+    public static class ModelStateDictionaryUtility
+    {
+        /// <summary>
+        /// Clears <see cref="ModelStateDictionary"/>.
+        /// </summary>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/>.</param>
+        /// <param name="type">The <see cref="Type"/> of object whose properties entries have to be cleared from
+        /// <see cref="ModelStateDictionary"/>.</param>
+        public static void ClearModelStateDictionary([NotNull] ModelStateDictionary modelState, [NotNull] Type type)
+        {
+            var properties = type.GetProperties();
```

Comment:
We really really really want to avoid just dropping down to reflection, even when it helps us solve a problem - because it creates another one... now we have reflection code that we have to implement caching for.

What's probable the right change here is to do a prefix check for the model state key that we're clearing. That will also solve the problem for models that have indexing (array, dictionary, etc).